test/unit/select_dropped_indexes_test.lua

```diff
@@ -0,0 +1,118 @@
+local select_plan = require('crud.select.plan')
+
+local select_conditions = require('crud.select.conditions')
+local cond_funcs = select_conditions.funcs
+
+local t = require('luatest')
+local g = t.group('select_dropped_indexes')
+
+local helpers = require('test.helper')
+
+g.before_all = function()
+    helpers.box_cfg()
+
+    local customers = box.schema.space.create('customers', {
+        format = {
+            {'id', 'unsigned'},
+            {'bucket_id', 'unsigned'},
+            {'name', 'string'},
+            {'age', 'unsigned'},
+            {'number_of_pets', 'unsigned'},
+            {'cars', 'array'},
+        },
+        if_not_exists = true,
+    })
+
+    customers:create_index('index1', {
+        type = 'TREE',
+        parts = {'id'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index2', {
+        parts = {'bucket_id'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('index3', {
+        type = 'TREE',
+        parts = {'age'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('index_dropped1', {
+        type = 'HASH',
+        parts = {'name'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index_dropped2', {
+        type = 'HASH',
+        parts = {'age'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index6', {
+        type = 'TREE',
+        parts = {'name'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers.index.index_dropped1:drop()
+    customers.index.index_dropped2:drop()
```

Comment:
It would be great to add some check to show that `#customers.index == 3` but actually there are 4 indexes.